llvm/lib/Target/RISCV/RISCVFoldMasks.cpp

```diff
@@ -62,6 +69,54 @@ char RISCVFoldMasks::ID = 0;
 
 INITIALIZE_PASS(RISCVFoldMasks, DEBUG_TYPE, "RISC-V Fold Masks", false, false)
 
+// If an AVL is a VLENB that's possibly scaled to be equal to VLMAX, convert it
+// to the VLMAX sentinel value.
+bool RISCVFoldMasks::convertToVLMAX(MachineInstr &MI) const {
+  if (!RISCVII::hasVLOp(MI.getDesc().TSFlags) ||
```

Comment:
Maybe we need a `isRVVPseudo` helper, I think it can be very useful.

Reply:
Yeah, I think RISCVInsertVSETVLI relies on hasSEWOp to check if an instruction is a pseudo.

Reply:
I think we could use `RISCVVPseudosTable::getPseudoInfo(Opc)` to check if it's a pseudo.

For here do we need to use it, or is it ok to just check for VL and SEW ops?

Reply:
Looking up table may be slow, I think it's OK to just check VL/SEW.